projects/packages/forms/src/dashboard/inbox/dataviews/actions.js

```diff
@@ -27,6 +27,21 @@ export const viewActionModal = {
 	},
 };
 
+export const editFormAction = {
+	id: 'edit-form',
+	label: __( 'Edit form', 'jetpack-forms' ),
+	icon: <Icon icon={ backup } />,
+	isEligible: item => !! item?.edit_form_url,
+	supportsBulk: false,
+	async callback( items ) {
+		const [ item ] = items;
+		if ( item?.edit_form_url ) {
+			const url = new URL( item.edit_form_url, window.location.origin );
+			window.open( url.toString(), '_blank', 'noopener' );
```

Comment:
Opening not in a new tab seems fine, no? New tabs tend to be disorienting, while in same tab you can return with back button. Especially on mobile.